scripts/start.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
+
+# Cleanup
+/usr/bin/nsenter -m/proc/1/ns/mnt -- fusermount -u /var/lib/lxc/lxcfs 2> /dev/null || true
+/usr/bin/nsenter -m/proc/1/ns/mnt -- [ -L /etc/mtab ] || \
+        sed -i "/^lxcfs \/var\/lib\/lxc\/lxcfs fuse.lxcfs/d" /etc/mtab
+
+# Prepare
+mkdir -p /usr/local/lib/lxcfs /var/lib/lxc/lxcfs
+
+# Update lxcfs
+cp -f /lxcfs/lxcfs /usr/local/bin/lxcfs
+cp -f /lxcfs/liblxcfs.so /usr/local/lib/lxcfs/liblxcfs.so
+
+
+# Mount
+exec /usr/bin/nsenter -m/proc/1/ns/mnt /usr/local/bin/lxcfs /var/lib/lxc/lxcfs/ &
```

Comment:
What if multiple debug-agents are launched simultaneously?

Reply:
good point, but is it really necessary to launched multiple debug-agents in one host? Should we check it before we run another debug-agent?

Reply:
some test cases listed below:

client logs: 
```
[root@iZuf6f9dx8ur1ouh5sb18gZ debug_lxcfs]$kubectl-debug --agentless=true --port-forw ard debugtest
Agent Pod info: [Name:debug-agent-pod-36d8240f-06a1-11ea-ae90-00163e06d596, Namespace:default, Image:registry.cn-hangzhou.aliyuncs.com/huya_zhangyi/debug-agent:0.0.2, HostPort:10027, ContainerPort:10027]
Waiting for pod debug-agent-pod-36d8240f-06a1-11ea-ae90-00163e06d596 to run...
set container procfs correct true..
pulling image nicolaka/netshoot:latest...
latest: Pulling from nicolaka/netshoot
Digest: sha256:8b020dc72d8ef07663e44c449f1294fc47c81a10ef5303dc8c2d9635e8ca22b1
Status: Image is up to date for nicolaka/netshoot:latest
starting debug container...
container created, open tty...

[root@iZuf6f9dx8ur1ouh5sb18gZ ~]$ kubectl-debug --agentless=true --port-forw ard debugtest
Agent Pod info: [Name:debug-agent-pod-9fdce875-06a1-11ea-afa5-00163e06d596, Namespace:default, Image:registry.cn-hangzhou.aliyuncs.com/huya_zhangyi/debug-agent:0.0.2, HostPort:10028, ContainerPort:10027]
Waiting for pod debug-agent-pod-9fdce875-06a1-11ea-afa5-00163e06d596 to run...
set container procfs correct true ...
pulling image nicolaka/netshoot:latest...
latest: Pulling from nicolaka/netshoot
Digest: sha256:8b020dc72d8ef07663e44c449f1294fc47c81a10ef5303dc8c2d9635e8ca22b1
Status: Image is up to date for nicolaka/netshoot:latest
starting debug container...
container created, open tty...


```
debug-agent logs: 
```
root@iZuf6f9dx8ur1ouh5sb18hZ:~# docker logs -f 96e41922affa
2019/11/14 05:40:09 Listening on 0.0.0.0:10027
mount namespace: 5
hierarchies:
  0: fd:   6: cpu,cpuacct
  1: fd:   7: memory
  2: fd:   8: pids
  3: fd:   9: devices
  4: fd:  10: cpuset
  5: fd:  11: perf_event
  6: fd:  12: freezer
  7: fd:  13: net_cls,net_prio
  8: fd:  14: blkio
  9: fd:  15: hugetlb
 10: fd:  16: name=systemd
 11: fd:  17: unified
2019/11/14 05:40:09 receive debug request
2019/11/14 05:40:09 Accept new debug reqeust:
	 target container: d51d185b37a4ca1cb30377f0b7b585470f8e5ff5bf3c6bb57f2d50151b9de825
	 image: nicolaka/netshoot:latest
	 command: [/bin/bash -l]
2019/11/14 05:40:09 remount lxcfs when the rootdir of lxcfs of target container has been mounted.

^C
root@iZuf6f9dx8ur1ouh5sb18hZ:~# docker logs -f 1434b5b437ea
2019/11/14 05:43:03 Listening on 0.0.0.0:10027
mount namespace: 5
hierarchies:
  0: fd:   6: cpu,cpuacct
  1: fd:   7: memory
  2: fd:   8: pids
  3: fd:   9: devices
  4: fd:  10: cpuset
  5: fd:  11: perf_event
  6: fd:  12: freezer
  7: fd:  13: net_cls,net_prio
  8: fd:  14: blkio
  9: fd:  15: hugetlb
 10: fd:  16: name=systemd
 11: fd:  17: unified
2019/11/14 05:43:04 receive debug request
2019/11/14 05:43:04 Accept new debug reqeust:
	 target container: d51d185b37a4ca1cb30377f0b7b585470f8e5ff5bf3c6bb57f2d50151b9de825
	 image: nicolaka/netshoot:latest
	 command: [/bin/bash -l]
2019/11/14 05:43:04 remount lxcfs when the rootdir of lxcfs of target container has been mounted.
```